mpSession/mpSession.cpp

```diff
@@ -0,0 +1,237 @@
+/* Copyright (C) 2005-2015 Massachusetts Institute of Technology
+%
+%  This program is free software; you can redistribute it and/or modify
+%  it under the terms of the GNU General Public License as published by
+%  the Free Software Foundation; either version 2, or (at your option)
+%  any later version.
+%
+%  This program is distributed in the hope that it will be useful,
+%  but WITHOUT ANY WARRANTY; without even the implied warranty of
+%  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%  GNU General Public License for more details.  %
+%  You should have received a copy of the GNU General Public License
+%  along with this program; if not, write to the Free Software Foundation,
+%  Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+*/
+
+#include "mpSession.hpp"
+#include "meep-ctl-const.hpp"
+
+using namespace meep;
+using namespace meepSession;
+
+/*--------------------------------------------------------------*/
+/* the next several routines create and initialize instances of */
+/* structures defined in ctl-io.h; they replicate the behavior  */
+/* of (define-class ...) statements in libctl/xx.scm files, and */
+/* calling them from C++/python codes is the equivalent of      */
+/* saying (make ...) in .ctl scripts.                           */
+/*--------------------------------------------------------------*/
+
+/* (define-class medium material-type ...                       */
+ctlio::material_type *make_dielectric(double epsilon)
+ { 
+   ctlio::material_type *mt = (ctlio::material_type *)malloc(sizeof(ctlio::material_type));
```

Comment:
I tend to use `malloc` for `struct`s and `new` for `class`es, but to be honest I don't know if this is standard practice.